http-gateway/web/src/components/copy-box/index.js

```diff
@@ -0,0 +1,93 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { injectIntl } from 'react-intl'
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
+import Tooltip from 'react-bootstrap/Tooltip'
+import { v4 as uuidv4 } from 'uuid'
+
+import { copyToClipboard } from '@/common/utils'
+import { messages as t } from './copy-box-i18n'
+
+class _CopyBox extends PureComponent {
```

Comment:
how come not a functional component?

Reply:
Mixing things up for fun! :) No real reason really, just felt like this could be a Class component.